api/nodes/serializers.py

```diff
@@ -130,6 +131,125 @@ def update(self, instance, validated_data):
         return instance
 
 
+class NodeContributorsSerializer(UserSerializer):
+
+    id = ser.CharField(source='_id')
+    permissions = ser.ListField(read_only=True)
+    bibliographic = ser.BooleanField(initial=True, help_text='Whether the user will be included in citations for '
+                                                               'this node or not')
+    permission = ser.ChoiceField(choices=['read', 'write', 'admin'], initial='write', write_only=True)
+    local_filterable = frozenset(['permission', 'bibliographic'])
+    filterable_fields = frozenset.union(UserSerializer.filterable_fields, local_filterable)
+
+    fullname = ser.CharField(read_only=True, help_text='Display name used in the general user interface')
```

Comment:
Yeah, we probably had this discussion before. You *might* be able to iterate through the field definitions and set read_only to true. If you could do that in a generic way, it would be good. Another way would be to make the actual user a related field rather than duplicating the info. That way people could grab the user info if need be, or use your upcoming include functionality once that's ready.